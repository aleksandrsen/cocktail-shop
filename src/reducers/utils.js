export const resourceStartRecord = () => ({
    loading: false,
    loaded: false,
    error: null,
    entities: {}
});

export const arrToObj = arr => {
    return arr.reduce((init, item) => {
        init[item.id] = item;
        return init;
    }, {});
};