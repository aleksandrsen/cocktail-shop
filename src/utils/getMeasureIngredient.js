export default function getMeasureIngredient(obj) {
    let measure = [];
    if (!obj.strMeasure1) return null;
    for (let key in obj) {
        if (key.slice(0, -1) === "strMeasure" && obj[key] !== null) {
            measure.push(obj[key]);
        }
    }
    return measure;
}