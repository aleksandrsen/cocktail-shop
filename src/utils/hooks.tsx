import { RefObject, useCallback, useEffect, useState } from "react";
import { isInViewport } from "./index";
import {useLocation} from "react-router-dom";

export const useMedia = (
  queries: string[],
  values: number[],
  defaultValue: number
) => {
  const mediaQueryLists = queries.map((q) => window.matchMedia(q));

  const getValue = useCallback(() => {
    const index = mediaQueryLists.findIndex((mql) => mql.matches);
    return values[index] ? values[index] : defaultValue;
  }, [mediaQueryLists, values, defaultValue]);

  const [value, setValue] = useState(getValue);

  useEffect(() => {
    const handler = () => setValue(getValue);
    mediaQueryLists.forEach((mql) => mql.addListener(handler));
    return () => mediaQueryLists.forEach((mql) => mql.removeListener(handler));
  }, [getValue, mediaQueryLists]);

  return value;
};

export const useIsShowAnimation = (ref: RefObject<any>, name?: string) => {
  const [isShow, setIsShow] = useState(true);

  const location = useLocation();

  useEffect(() => {
    if (ref?.current) {

      const windowOffset = window.pageYOffset;

      console.log(windowOffset, "windowOffset")
      console.log(name, isInViewport(ref.current), ref.current.offsetTop, windowOffset)

      if (isInViewport(ref.current) || ref.current.offsetTop <= windowOffset) {
        setIsShow(false);
        console.log("inside")
      }

    }
  }, [ref?.current, location]);

  return isShow;
};
