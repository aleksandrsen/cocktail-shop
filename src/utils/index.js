export const cutTextContent = (text, maxLength) => {
    if (text.length > maxLength) {
        return text.slice(0, maxLength - 3) + "..."
    }
    return text;
}

export const formatDate = (dateString, optionsObj) => {
    const defaultOptions = {
        year: "numeric",
        month: "short",
        day: "numeric",
        hour: "numeric",
        minute: "numeric"
    };

    const date = new Date(Date.parse(dateString));
    const formatter = new Intl.DateTimeFormat("en", optionsObj || defaultOptions);
    return formatter.format(date)
}