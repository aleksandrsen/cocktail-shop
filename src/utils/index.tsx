import React, { JSXElementConstructor, useEffect } from "react";
import { useLocation } from "react-router-dom";

export const ScrollTop = (): null => {
  const { pathname } = useLocation();

  useEffect(() => {
    window.scrollTo(0, 0);
  }, [pathname]);
  return null;
};

export const cutTextContent = (text: string, maxLength: number): string => {
  if (text.length > maxLength) return text.slice(0, maxLength - 3) + "...";
  return text;
};

export function getSkeletons(
  length: number,
  SkeletonComponent: JSXElementConstructor<any>,
  props: object = {}
): JSX.Element[] {
  return new Array(length)
    .fill(1)
    .map((num, idx) => <SkeletonComponent key={`${num}${idx}`} {...props} />);
}

type DateOptionItem = "2-digit" | "numeric";

type DateOptionsObj = {
  year?: DateOptionItem;
  month?: DateOptionItem | "narrow" | "short" | "long";
  weekday?: "narrow" | "short" | "long";
  day?: DateOptionItem;
  hour?: DateOptionItem;
  minute?: DateOptionItem;
};

export const formatDate = (
  date: string | Date | undefined,
  optionsObj?: DateOptionsObj
): string => {
  if (!date) return "";
  const defaultOptions = {
    year: "numeric",
    month: "short",
    day: "numeric",
    hour: "numeric",
    minute: "numeric",
  };

  const date_ = typeof date === "string" ? new Date(Date.parse(date)) : date;
  const formatter = new Intl.DateTimeFormat("en", optionsObj || defaultOptions);
  return formatter.format(date_);
};

export function searchByFields<T extends object, U extends keyof T>(
  arrayOfObj: T[],
  searchValue: string,
  keyFields: U[]
): T[] {
  const result: T[] = [];
  const term = searchValue.replace(",", " ").toLowerCase();

  arrayOfObj.forEach((obj: T) => {
    const descr = keyFields
      .reduce((arr: T[U][], field) => {
        arr.push(obj[field]);
        return arr;
      }, [])
      .join(" ");

    if (descr.toLowerCase().includes(term)) result.push(obj);
  });
  return result;
}

export const validateEmail = (str: string): boolean =>
  !/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i.test(str);

export const getTotalValue = <T extends object, K extends keyof T>(
  arr: T[],
  field: K
): number => {
  type Val = {
    [T[K]]: number
  }
  return arr.reduce((sum, item) => sum + item[field], 0)
};
