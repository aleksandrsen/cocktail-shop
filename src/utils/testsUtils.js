import { applyMiddleware, createStore } from "redux";
import { sagaMiddleware } from "../store";
import rootReducer from "../reducers";

export const testStore = (initialState) => {
  const createStoreWithMiddleware = applyMiddleware(sagaMiddleware)(
    createStore
  );
  return createStoreWithMiddleware(rootReducer, initialState);
};

export const findByTestAttr = (component, attr) =>
  component.find(`[data-test='${attr}']`);
