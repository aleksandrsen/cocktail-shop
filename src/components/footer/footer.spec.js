import React from "react";
import Footer from "./footer";
import { shallow } from "enzyme";
import { findByTestAttr } from "../../utils/testsUtils";

const setUp = () => shallow(<Footer />);

describe("Should render Footer component without errors", () => {
  let component;
  beforeEach(() => {
    component = setUp();
  });

  it("Should render Footer component", () => {
    expect(component).toMatchSnapshot();
  });

  it("Should render Footer component wrapper", () => {
    const wrapper = findByTestAttr(component, "mainFooter");
    expect(wrapper.length).toBe(1);
  });
});
