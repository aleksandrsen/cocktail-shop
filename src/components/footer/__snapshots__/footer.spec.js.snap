// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Footer component without errors Should render Footer component 1`] = `
<footer
  className="mainFooter"
  data-test="mainFooter"
>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col col-4 col-md-6 col-sm-12"
      >
        <div
          className="mainFooter__contacts"
        >
          <h4
            className="mainFooter__title"
          >
            Contact us
          </h4>
          <div
            className="mainFooter__contactItem"
          >
            <span>
              Address: 
            </span>
            New York, 345 Park AveNY 10154, USA
          </div>
          <div
            className="mainFooter__contactItem"
          >
            <span>
              Email: 
            </span>
            pubOnHub@gmail.com
          </div>
          <div
            className="mainFooter__contactItem"
          >
            <span>
              Phone: 
            </span>
            +1 245 123 222
          </div>
        </div>
      </div>
      <div
        className="col col-4 col-md-12"
      >
        <img
          alt="logo"
          src="logo.png"
        />
      </div>
      <div
        className="col col-4 col-md-6 col-sm-12"
      >
        <div
          className="mainFooter__about"
        >
          <h4
            className="mainFooter__title"
          >
            About story
          </h4>
          <p>
            Lorem ipsum dolor sit amet, consectetur adipisicing elit. Aliquam aspernatur autem beatae delectus dolores dolorum error est ex inventore laudantium nam numquam optio, quod sint soluta, totam vitae voluptate voluptates.
          </p>
          <span
            className="hub"
          >
            Hub on pubs
          </span>
        </div>
      </div>
    </div>
    <Memo(SocialNetworks) />
    <div
      className="mainFooter__copyright"
    >
      <svg>
        <use
          xlinkHref="#footer-copyright"
        />
      </svg>
      2021
       - Hub on pubs - All Rights Reserved
    </div>
  </div>
</footer>
`;
